cfme/tests/containers/test_pause_resume_workers.py

```diff
@@ -52,10 +55,17 @@ def test_pause_and_resume_provider_workers(appliance, provider):
     ems_worker_state = check_ems_state_in_diagnostics(appliance, provider)
     assert not ems_worker_state, "Diagnostics shows that workers are running after pause provider"
 
+    @request.addfinalizer
+    def _finalize():
+        if not provider.is_provider_enabled:
+            view = navigate_to(provider, "Details")
+            view.toolbar.configuration.item_select(provider.resume_provider_text, handle_alert=True)
+            wait_for(lambda: not check_ems_state_in_diagnostics(appliance, provider))
```

Comment:
optional:
can we avoid `wait_for` in `finalizer`? Its part of tear down and looks like, we are trying to validate (waiting for ems_state) in finalizer.